src/sentry/lang/native/unreal.py

```diff
@@ -16,6 +16,35 @@
 ''')
 
 
+def is_unreal_event(data):
+    """Whether this event is an Unreal crash that should be processed in
+    enhancers. For the legacy codepath this still returns False."""
+    return get_path(data, 'contexts', 'unreal', 'type') == 'unreal'
+
+
+def reprocess_unreal_crash(data):
+    context = get_path(data, 'contexts', 'unreal')
+    portable_call_stack = get_path(context, 'portable_call_stack')
+
+    if not portable_call_stack:
+        return
+
+    images = get_path(data, 'debug_meta', 'images', filter=True, default=())
+    exception = get_path(data, 'exception', 'values', 0)
+
+    frames = parse_portable_callstack(portable_call_stack, images)
+    if not frames:
+        return
+
+    if exception:
+        exception['mechanism'] = {'type': 'unreal', 'handled': False, 'synthetic': True}
+        exception['stacktrace'] = {'frames': frames}
+    else:
+        data['stacktrace'] = {'frames': frames}
```

Comment:
It might be nice to create a synthetic exception in this case and set the mechanism. Otherwise, it might look arbitrary to the user why in some cases the mechanism would be set and in some not.

Reply:
Done. Please review snapshot diffs carefully